algebird-core/src/main/scala/com/twitter/algebird/Monoid.scala

```diff
@@ -47,8 +48,14 @@ trait Monoid[@specialized(Int, Long, Float, Double) T] extends Semigroup[T] {
       None
     }
   }
-  // Override this if there is a more efficient means to implement this
-  def sum(vs: TraversableOnce[T]): T = sumOption(vs).getOrElse(zero)
+  override def sum(vs: TraversableOnce[T]): T = sumOption(vs).getOrElse(zero)
+
+  /**
+   * These are from algebra.Monoid
+   */
+  override def additive: AMonoid[T] = this
+  override def empty: T = zero
+  override def combineAll(t: TraversableOnce[T]): T = sum(t)
```

Comment:
You aren't missing anything. It is totally a naming thing.

On Thursday, June 9, 2016, Alex Levenson <notifications@github.com> wrote:

> In algebird-core/src/main/scala/com/twitter/algebird/Monoid.scala
> <https://github.com/twitter/algebird/pull/523#discussion_r66557027>:
>
> > @@ -47,8 +48,14 @@ trait Monoid[@specialized(Int, Long, Float, Double) T] extends Semigroup[T] {
> >        None
> >      }
> >    }
> > -  // Override this if there is a more efficient means to implement this
> > -  def sum(vs: TraversableOnce[T]): T = sumOption(vs).getOrElse(zero)
> > +  override def sum(vs: TraversableOnce[T]): T = sumOption(vs).getOrElse(zero)
> > +
> > +  /**
> > +   * These are from algebra.Monoid
> > +   */
> > +  override def additive: AMonoid[T] = this
> > +  override def empty: T = zero
> > +  override def combineAll(t: TraversableOnce[T]): T = sum(t)
>
> sorry, still don't understand. Is this just about naming?
> I thought a monoid was:
> 1) A set of elements, E
> 2) A (closed) binary operator over the elements in E
> 3) An identity element present in E
>
> You can call the operator "plus" and the identity "zero" or you can call
> it "times" and "one" but that doesn't really mean anything in terms of how
> it works right? Or am I missing something and there's a functional
> difference between these two?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/twitter/algebird/pull/523/files/053fd81dafbd88982cad3b9b5c8839ab16ceb399#r66557027>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAEJdhyBqldBPjEvwCIrlhP-IHQI5yJXks5qKNaxgaJpZM4Ieuhu>
> .
>


-- 
P. Oscar Boykin, Ph.D. | http://twitter.com/posco | http://pobox.com/~boykin